utils.groovy

```diff
@@ -516,26 +516,32 @@ def build_artifacts(pipecfg, stream, basearch) {
 
 def get_registry_repos(pipecfg, stream, version) {
     def registry_repos = pipecfg.registry_repos ?: [:]
-    // merge top-level registry_repos with stream-specific bits
-    registry_repos += pipecfg.streams[stream].additional_registry_repos ?: [:]
+    // get stream-specific bits additional repos
+    def additional_repos = pipecfg.streams[stream].additional_registry_repos ?: [:]
+    for (repo in ['oscontainer', 'legacy_oscontainer', 'extensions', 'kubevirt']) {
+        //merge top-level registry_repos with addtional repos
+        registry_repos[repo] += additional_repos[repo] ?: []
+     }
     for (repo in (registry_repos.keySet() as List)) {
```

Comment:
suggest to change this to repo_id (i.e. `oscontainer`, `extensions`, etc). It's just the key in the hash. 
```suggestion
    for (repo_id in (registry_repos.keySet() as List)) {
```